pilot/pkg/proxy/envoy/proxy.go

```diff
@@ -140,9 +143,18 @@ func (e *envoy) Run(config interface{}, epoch int, abort <-chan error) error {
 	if err := cmd.Start(); err != nil {
 		return err
 	}
+	// record envoy process number
+	e.processNum.Inc()
 	done := make(chan error, 1)
 	go func() {
-		done <- cmd.Wait()
+		err := cmd.Wait()
+		e.processNum.Dec()
```

Comment:
I think you can just save the result of `Dec`, rather than calling `Load` below.

Reply:
good